tools/clang/unittests/HLSLExec/ShaderOpArith.xml

```diff
@@ -4184,7 +4184,29 @@ void MSMain(uint GID : SV_GroupIndex,
             const uint32_t OutNum = NUM;
           #endif
 
-          #if IS_UNARY_OP
+          #if TEST_ARRAY_OPERATOR_STATIC_ACCESS
+            uint IndexList[6] = {0, OutNum - 1, 1, OutNum - 2, OutNum / 2, OutNum / 2 + 1};
+            
+            vector<OUT_TYPE, OutNum> OutputVector = 0;
+            uint End = min(OutNum, 6);
+
+            [unroll]for(uint i = 0; i < 6; ++i) {
+              OutputVector[IndexList[i]] = Input1[IndexList[i]];
+              uint index = (uint)(IndexList[i]);
+            }
+          #elif TEST_ARRAY_OPERATOR_DYNAMIC_ACCESS
+            // Input2 will always be 0. This is added so the compiler cannot optimize the array access.
+            uint Modifier = (uint) Input2[0];
+            uint IndexList[6] = {0 + Modifier, OutNum - 1 + Modifier, 1 + Modifier, OutNum - 2 + Modifier, OutNum / 2 + Modifier, OutNum / 2 + 1 + Modifier};
```

Comment:
Typically you can use the name `Zero` for clarity. Also, if you add this `Zero` to the initialization of `index` below instead of the init list, the code would be a bit simpler, and equivalent. Then it's easier to compare `IndexList` between static/dynamic cases. If you had a common location, you could even share the code.